aws-analytics-pinpoint/src/test/java/com/amplifyframework/analytics/pinpoint/AWSPinpointAnalyticsPluginBehaviorTest.kt

```diff
@@ -134,4 +148,69 @@ class AWSPinpointAnalyticsPluginBehaviorTest {
         verify(exactly = 1) { autoEventSubmitterMock.stop() }
         verify(exactly = 1) { autoSessionTrackerMock.stopSessionTracking(context.applicationContext as Application) }
     }
+
+    @Test
+    fun `test identify user`() {
+        every { sharedPrefsUniqueIdService.getUniqueId() }.answers { "UNIQUE_ID" }
+        every { targetingClientMock.currentEndpoint() }.answers {
+            EndpointProfile(
+                pinpointNotificationClient,
+                sharedPrefsUniqueIdService,
+                androidAppDetails,
+                androidDeviceDetails,
+                ApplicationProvider.getApplicationContext()
+            )
+        }
+        val actualEndpoint = slot<EndpointProfile>()
+        every { targetingClientMock.updateEndpointProfile(capture(actualEndpoint)) } returns Unit
+
+        val properties =
+            AnalyticsProperties.builder()
+                .add("key1", "String1")
+                .add("key2", 1.0)
+                .add("key3", true)
+                .add("key4", 1)
+                .build()
+        val userProfile = AWSPinpointUserProfile.builder()
+            .email("test@test.com")
+            .name("test")
+            .userAttributes(properties)
+            .build()
+        awsPinpointAnalyticsPluginBehavior.identifyUser(
+            "USER_ID",
+            userProfile
+        )
+
+        verify(exactly = 1) { targetingClientMock.updateEndpointProfile(any()) }
+        val expectedUserAttributes =
```

Comment:
can you place expectations at the top of test so that GIVEN/WHEN/THEN flow is not broken